test/Swashbuckle.AspNetCore.SwaggerGen.Test/XmlComments/XmlMethodInfoExtensionsTests.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System.Linq;
+using System.Reflection;
+using Swashbuckle.AspNetCore.SwaggerGen.Test.Fixtures;
+using Xunit;
+
+namespace Swashbuckle.AspNetCore.SwaggerGen.Test;
+
+public class XmlMethodInfoExtensionsTests
+{
+    [Theory]
+    [MemberData(nameof(DifferentMethodsSignatures))]
+    public void DifferentMethodsSignatures_ShouldBeExpected(MethodInfo methodInfo)
+    {
+        var underlyingGenericMethod = methodInfo.GetUnderlyingGenericTypeMethod();
+        Assert.NotNull(underlyingGenericMethod);
+        var underlyingGenericMethodParameters = underlyingGenericMethod.GetParameters();
+        Assert.NotEmpty(underlyingGenericMethodParameters);
+        Assert.Equal(methodInfo.GetParameters().Select(s => s.Name), underlyingGenericMethodParameters.Select(s => s.Name));
+        Assert.Equal(methodInfo.ReturnParameter.ParameterType, underlyingGenericMethod.ReturnParameter.ParameterType);
```

Comment:
I think it would be better to explicitly return the expected names and types as part of the theory data. Otherwise you're effectively using the same logic as the code you're testing to compute what the expected results should be.

Reply:
The parameters names are not checked in the GetUnderlyingGenericTypeMethod, I am checking that the parameters are the expetect(That are the ones that come from the methodInfo).
I can only check the parametersNames in this case, because the parameterTypes in the case of the underlyingGenericMethod are generic

Reply:
@martincostello I keep it open in case you need a change here

Reply:
What I'm getting at is that you're dynamically computing what you expect the result to be.

Instead of would be better for you to explicitly reference the methods to get the `MethodInfo`s and then hard-code what you expect the names and types to be, so for example:

```csharp
[firstMethodInfo, ["t", "cancellationToken"], typeof(byte)],
[secondMethodInfo, ["key", "t"], typeof(sbyte)],
// etc.
```

This would then go with asserts like:

```csharp
Assert.Equal(methodInfo.GetParameters().Select(s => s.Name), expectedParameterNames);
Assert.Equal(methodInfo.ReturnParameter.ParameterType, expectedReturnParameterType);
```

That's the general gist anyway.